From 2c82781c540c5ca71ad03e25dc9414af09553db2 Mon Sep 17 00:00:00 2001
From: Quanyang Wang <quanyang.wang@windriver.com>
Date: Fri, 3 Jul 2015 07:56:25 +0000
Subject: [PATCH 848/987] net: dpaa: delete redundant free function from usdpaa
 driver

In freescale usdpaa driver, the variable "proxy_dev" is allocated by
function "devm_kzalloc". When the probe function fails, the system will
automatically free the "proxy_dev" and should not use kfree to do
the release operation, else it will has the kmemleak warning as below:

kmemleak: Found object by alias at 0xe8d672e0
CPU: 1 PID: 1 Comm: swapper/0 Not tainted 3.14.29ltsi-WR7.0.0.0_standard #2
Call Trace:
[ef5f5ba0] [c0008e1c] show_stack+0xfc/0x1c0 (unreliable)
[ef5f5bf0] [c0a50720] dump_stack+0x7c/0xdc
[ef5f5c10] [c01ddc60] find_and_get_object+0x220/0x240
[ef5f5c40] [c01de8d4] delete_object_full+0x24/0x70
[ef5f5c60] [c01c60b8] kfree+0x1b8/0x3e0
[ef5f5ca0] [c0a55508] dpa_eth_proxy_remove+0x2c/0x50
[ef5f5cc0] [c06ff424] dpaa_eth_proxy_probe+0x224/0x230
[ef5f5d10] [c058cc14] platform_drv_probe+0x34/0x80
[ef5f5d30] [c058a658] really_probe+0x88/0x280
[ef5f5d60] [c058a9c8] __driver_attach+0xe8/0xf0
[ef5f5d80] [c058807c] bus_for_each_dev+0x7c/0xd0
[ef5f5db0] [c0589b18] bus_add_driver+0x188/0x260
[ef5f5de0] [c058b304] driver_register+0x94/0x150
[ef5f5e00] [c0d39a80] dpa_proxy_load+0x44/0x88
[ef5f5e20] [c0002458] do_one_initcall+0x158/0x1d0
[ef5f5e90] [c0d07a5c] kernel_init_freeable+0x1d8/0x2a8
[ef5f5f20] [c0002c40] kernel_init+0x20/0x110
[ef5f5f40] [c0012ef0] ret_from_kernel_thread+0x5c/0x64
kmemleak: Object 0xe8d672d0 (size 32):
kmemleak:   comm "swapper/0", pid 1, jiffies 4294893436
kmemleak:   min_count = 1
kmemleak:   count = 0
kmemleak:   flags = 0x1
kmemleak:   checksum = 0
kmemleak:   backtrace:

Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 drivers/net/ethernet/freescale/dpa/dpaa_eth_proxy.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/drivers/net/ethernet/freescale/dpa/dpaa_eth_proxy.c b/drivers/net/ethernet/freescale/dpa/dpaa_eth_proxy.c
index 682e2b2..560a01e 100644
--- a/drivers/net/ethernet/freescale/dpa/dpaa_eth_proxy.c
+++ b/drivers/net/ethernet/freescale/dpa/dpaa_eth_proxy.c
@@ -312,8 +312,6 @@ static int __cold dpa_eth_proxy_remove(struct platform_device *of_dev)
 	struct device *dev = &of_dev->dev;
 	struct proxy_device *proxy_dev = dev_get_drvdata(dev);
 
-	kfree(proxy_dev);
-
 	dev_set_drvdata(dev, NULL);
 
 	return 0;
-- 
1.9.1

